[ENH] BEP001 - Entity-linked file collections (#688)
bids-standard/bids-specification: src/02-common-principles.md

specified is available in the [entity table](./99-appendices/04-entity-table.md)
in the appendix.

### Entity-linked file collections

> A set of files that are related to each other based on a repetitive acquisition of
> sequential data by changing acquisition parameters one (or multiple) at a time and/or by
> being inherent components of the same data.

Author: Better to use a full sentence instead of a dictionary-like fragment:

```suggestion
An entity-linked file collection is a set of files that are related to each other
based on a repetitive acquisition of sequential data
by changing acquisition parameters one (or multiple) at a time
or by being inherent components of the same data.
```

I've dropped the `and/or` for `or`. I'm not sure being an explicitly inclusive "or" gains clarity.

> The suffix serves as an identifier for these
> collections, holding a logical grouping of constituent files.

Author: I don't quite understand this sentence. I think "holding" means that the suffix indicates which set of files to expect. What about:

```suggestion
Entity-linked collections are identified by a common suffix,
indicating the group of files that should be considered a logical unit.
```

Within each collection, files MUST be distinguished from each other by at least one 
entity (for example, `echo`) that corresponds to an altered acquisition parameter
(`EchoTime`) or that defines a component relationship (for example, `part`).

> Note that these entities 
> MUST be described by the specification and the parameter changes they declare SHOULD NOT invalidate
> the definition of the accompanying suffix. For example, the use of the `echo` entity along with the 
> `T1w` suffix casts doubt on the validity of the identified contrast weighting.  

Author: Re-spacing for readability. Changing SHOULD NOT (not an RFC 2119 phrase) to MUST NOT, which I believe is the intent.

```suggestion
Note that these entities MUST be described by the specification and
the parameter changes they declare MUST NOT invalidate the definition of the accompanying suffix.
For example, the use of the `echo` entity along with the `T1w` suffix casts doubt on
the validity of the identified contrast weighting.  
```

> Provided the conditions above are satisfied, any suffix (such as `bold`) can identify an entity-linked file 
> collection, although certain suffixes are exclusive for this purpose (for example, `MP2RAGE`).

Author: ```suggestion
Provided the conditions above are satisfied,
any suffix (such as `bold`) can identify an entity-linked file collection,
although certain suffixes are exclusive for this purpose (for example, `MP2RAGE`).
```

Use cases concerning this convention are compiled in the
[file collections](./99-appendices/10-file-collections.md) appendix.

> This convention is mainly intended for but not
> limited to MRI modalities.

Author: ```suggestion
This convention is mainly intended for but not limited to MRI modalities.
```
